src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -2312,16 +2316,24 @@ private string GetLocaleInfoCore(LocaleStringData type, string? uiCultureName =
             // This is never reached but helps illinker statically remove dependencies
             if (GlobalizationMode.Invariant)
                 return null!;
-
-            return GlobalizationMode.UseNls ? NlsGetLocaleInfo(type) : IcuGetLocaleInfo(type, uiCultureName);
+#if TARGET_IOS
+            return GlobalizationMode.UseNls ? NlsGetLocaleInfo(type)
```

Comment:
`GlobalizationMode.UseNls` is always going to be false on iOS. It is Windows-specific property. You can omit it for iOS-specific code.